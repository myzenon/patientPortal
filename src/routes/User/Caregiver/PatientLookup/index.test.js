import React from 'react'
import { CaregiverPatientLookup } from './index'
import ReactGA from 'react-ga';
import moment from 'moment'

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

test('renders correctly', () => {
    const wrapper = render (
        <CaregiverPatientLookup />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('clicks next button | found = true', (done) => {
    const props = {
        push: (url) => {
            expect(url).toBe('/caregiver-patient-lookup-confirm')
            done()
        },
        callSyncAPIGateway: (name, get, post) => {
            expect(name).toBe('usersLookupInsuranceIdPost')
            expect(post).toEqual({
                insurance_id: 'INSURANCEID'
            })
            return Promise.resolve()
        },
        page: {
            apiData: {
                found: 'true'
            }
        },
        patients: [...mockObj.patients],
        savePageData: (data) => {
            expect(data).toEqual({
                insuranceId: 'INSURANCEID'
            })
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <CaregiverPatientLookup {...props} />
    )
    wrapper.find('input[name="insuranceId"]').simulate('change', {
        target: {
            name: 'insuranceId',
            value: 'insuranceId'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
})

test('clicks next button | found = false', (done) => {
    const props = {
        callSyncAPIGateway: (name, get, post) => {
            expect(name).toBe('usersLookupInsuranceIdPost')
            expect(post).toEqual({
                insurance_id: 'INSURANCEID'
            })
            return Promise.resolve()
        },
        page: {
            apiData: {
                found: 'false'
            }
        },
        patients: [...mockObj.patients],
        savePageData: (data) => {
            expect(data).toEqual({
                insuranceId: 'INSURANCEID'
            })
            return Promise.resolve()
        }
    }
    const wrapper = mount (
        <CaregiverPatientLookup {...props} />
    )
    wrapper.find('input[name="insuranceId"]').simulate('change', {
        target: {
            name: 'insuranceId',
            value: 'insuranceId'
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
    setTimeout(function () {
        expect(toJson(wrapper.render())).toMatchSnapshot()
        done()
    }, 0)
})

test('patient is given consent already | call fhirPatientsIdGet and fhirCmrGet API | go to /confirm-info', done => {
    const props = {
        callSyncAPIGateway: (name, get) => {
            if (name === 'fhirPatientsIdGet') {
                expect(get).toEqual({
                    id: 'b70f3cbe-ddd3-4ce3-96b2-914a609abd53'
                })
                return Promise.resolve({})
            }
            if (name === 'fhirCmrGet') {
                expect(get).toEqual({
                    patientId: 'b70f3cbe-ddd3-4ce3-96b2-914a609abd53'
                })
                return Promise.resolve({})
            }
            return Promise.reject({})
        },
        page: {
            apiData: {
                someData: 'someData'
            }
        },
        user: {...mockObj.patient},
        patients: [...mockObj.patients],
        setPatient: (patient) => {
            expect(patient).toEqual(props.page.apiData)
            return Promise.resolve({})
        },
        setCMR: (cmr) => {
            expect(cmr).toEqual(props.page.apiData)
            return Promise.resolve({})
        },
        callAPIGateway: (name, get, post) => {
            expect(name).toBe('fhirPatientsIdPut')
            // expect(post).toEqual({
            //     id: 'b70f3cbe-ddd3-4ce3-96b2-914a609abd53'
            //     // lastReviewDate: moment().utc().format('YYYY-MM-DDTHH:mm:ss')
            // })
            return Promise.resolve()
        },
        push: (url) => {
            expect(url).toBe('/confirm-info')
            done()
        }
    }
    const wrapper = mount (
        <CaregiverPatientLookup {...props} />
    )
    wrapper.find('input[name="insuranceId"]').simulate('change', {
        target: {
            name: 'insuranceId',
            value: mockObj.patients[0].insuranceId
        }
    })
    wrapper.find('button[type="submit"]').simulate('submit')
})
