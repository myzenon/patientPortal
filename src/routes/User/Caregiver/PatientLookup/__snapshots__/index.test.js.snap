// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicks next button | found = false 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <div
      class="instructionWrapper"
    >
      <p>
        Please enter the patient's information below
      </p>
    </div>
    <form
      class=""
    >
      <div
        class="inputTextWrapper"
      >
        <input
          class="inputTextError"
          name="insuranceId"
          placeholder="Enter Patient Health Plan Identification Number"
          type="text"
          value="insuranceId"
        />
        <div
          class="textError"
        >
          Uh oh! It looks like you might have entered the wrong ID, or we can't find your patient in our system. Give us a call and we'll help you sort it out!
        </div>
      </div>
      <br />
      <br />
       
      <br />
      <div
        class="helpWrapper"
      >
        <a
          href="/help-me-find-hpid"
        >
          Help me find my patient's Health Plan Identification Number
        </a>
      </div>
      <br />
      <p />
      <button
        class="button btn btn-secondary"
        type="submit"
      >
        NEXT
      </button>
    </form>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <div
      class="instructionWrapper"
    >
      <p>
        Please enter the patient's information below
      </p>
    </div>
    <form
      class=""
    >
      <div
        class="inputTextWrapper"
      >
        <input
          autofocus=""
          class="inputText"
          name="insuranceId"
          placeholder="Enter Patient Health Plan Identification Number"
          type="text"
          value=""
        />
      </div>
      <br />
      <br />
       
      <br />
      <div
        class="helpWrapper"
      >
        <a
          href="/help-me-find-hpid"
        >
          Help me find my patient's Health Plan Identification Number
        </a>
      </div>
      <br />
      <p />
      <button
        class="button btn btn-secondary"
        type="submit"
      >
        NEXT
      </button>
    </form>
  </div>
  <div
    class="footerWrapper container"
  >
    <div>
      <div
        class="wrapper"
      >
        <div
          class="stepActive"
        >
          Account
          <br />
          Setup
          <div
            class="stepNumberActive"
          >
            1
          </div>
        </div>
        <div
          class="step"
        >
          Medication
          <br />
          Review
          <div
            class="stepNumber"
          >
            2
          </div>
        </div>
        <div
          class="step"
        >
          Additional 
          <br />
          Medications
          <div
            class="stepNumber"
          >
            3
          </div>
        </div>
        <div
          class="step"
        >
          Health
          <br />
          Updates
          <div
            class="stepNumber"
          >
            4
          </div>
        </div>
        <div
          class="step"
        >
          Set
          <br />
          Appointment
          <div
            class="stepNumber"
          >
            5
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
